zerver/decorator.py

```diff
@@ -545,8 +545,16 @@ def _wrapped_func_arguments(request: HttpRequest, *args: Any, **kwargs: Any) ->
                                            client_name=full_webhook_client_name(webhook_client_name))
             except JsonableError as e:
                 return json_unauthorized(e.msg)
-            # Apply rate limiting
-            return rate_limit()(view_func)(request, profile, *args, **kwargs)
+            try:
+                # Apply rate limiting
+                return rate_limit()(view_func)(request, profile, *args, **kwargs)
+            except Exception as err:
+                if is_webhook or webhook_client_name is not None:
+                    request_body = request.POST.get('payload')
+                    if request_body is not None:
+                        log_exception_to_webhook_logger(request, profile,
+                                                        request_body=request_body)
+                raise err
```

Comment:
One thing that reading this makes me realize we should test is that a non-webhook view exception gets re-raised properly, since this is the main decorator for all of our views.  Can you add a unit test for that?

Reply:
@timabbott: Agreed! I just added the test! :)